scons/site_tools/helper.py

```diff
@@ -52,11 +52,11 @@
 filter = re.compile('^(?P<section>[.]\w+)\s*(?P<size>\d+)\s*(?P<addr>\d+)$')
 
 # Sections which will remain in the Flash
-flashSectionNames = ['.reset', '.fastcode', '.fastdata', '.text', '.rodata', '.data']
+flashSectionNames = ['.reset', '.vectors', '.fastcode', '.fastdata', '.text', '.rodata', '.data']
 
 # Sections which will be created in RAM or are copied from the Flash. In that
 # case the section will appear also in `flashSectionNames`.
-ramSectionNames = ['.vectors', '.fastcode', '.fastdata', '.data', '.bss', '.noinit']
+ramSectionNames = ['.vectors', '.fastcode', '.fastdata', '.data', '.bss', '.noinit', '.stack']
```

Comment:
Note, that this causes the displayed RAM usage to be higher from now on, since the stacks memory is bound and cannot be dynamically be used for heap storage.
Also no more stack and heap collisions. Ever.